synapse/lib/link.py

```diff
@@ -18,10 +18,15 @@
 READSIZE = 16 * s_const.mebibyte
 MAXWRITE = 64 * s_const.mebibyte
 
-async def connect(host, port, ssl=None, hostname=None, linkinfo=None):
+async def connect(host, port, ssl=None, hostname=None, linkinfo=None, linkcls=None):
```

Comment:
Is there a reason not to have linkcls=Link by default?

Classes are immutable objects and we'd be able to avoid the `if linkcls is None` check.

Reply:
Only because I'd have to move these functions down below where Link is declared. I'll move them.